packages/frontend/src/pages/Signup.tsx

```diff
@@ -63,6 +116,12 @@ const Signup: FC = () => {
     const allowPasswordAuthentication =
         !health.data?.auth.disablePasswordAuthentication;
 
+    useEffect(() => {
+        if (window && window.location.href.includes('?from=email')) {
+            setIsLinkFromEmail(true);
+        }
+    }, [inviteCode]);
```

Comment:
We should use `useLocation` so the hook can have the correct dependencies.
Eg:
```
const { search } = useLocation();
useEffect(() => {
         const searchParams = new URLSearchParams(search);
         const fromParam = searchParams.get('from');
         if (fromParam === 'email') {
               setIsLinkFromEmail(true);
        }
}, [search]);
```
Note that I didn't test this code, just wrote has an example.